Homeworks/Homework_2/task.py

```diff
@@ -0,0 +1,131 @@
+            self.data.extend(response.json()['results'])
+
+    def get_all_data_from_page(self):
+        return self.data
+
+    def get_data_about_movies(self, start, end, step):
+        return self.data[start:end:step]
+
+    def get_the_most_popular_movie(self):
+        return max(self.data, key=lambda x: x['popularity'])['original_title']
+
+    def find_movie(self, find_text):
+        return [film['original_title'] for film in self.data if film['overview'].find(find_text) > 0]
+
+    def print_genres(self):
+        return set(genre for film in self.data for genre in film['genre_ids'])
+
+    def delete_movies_with_genre(self, deleted_genre):
+        return [film['original_title'] for film in self.data if deleted_genre not in film['genre_ids']]
+
+    def get_most_popular_genre(self):
+        return Counter([genre for film in self.data for genre in film['genre_ids']]).most_common(1)[0][0]
+
+    def group_movies(self):
+        common_movies = [{film['original_title'], movie['original_title']} for i, film in enumerate(self.data) for genre in film['genre_ids'] for movie in self.data[i+1::] if genre in movie['genre_ids']]
+        # I left it here because the top line is a little bit too long, but it is written in one line :)
+        #
+        # for i, film in enumerate(self.data):
+        #     for genre in film['genre_ids']:
+        #         for movie in self.data[i+1::]:
+        #             if genre in movie['genre_ids']:
+        #                 common_movies.append({film['original_title'], movie['original_title']})
+        return common_movies
+
+    def replace_genre(self):
+        movies_copy = copy.deepcopy(self.data)
+        print(f'Default data:{self.data}')
+        movies_copy = list(map(change_genre_id, movies_copy))
+        print(f'Changed data:{movies_copy}')
+
+    def print_collection_of_structures(self):
+        for film in self.data:
+            example = {'Title': '',
+                       'Popularity': 0,
+                       'Score': 0,
+                       'Last_day_in_cinema': ''}
+            example['Title'] = film['original_title']
+            example['Popularity'] = round(film['popularity'], 1)
+            example['Score'] = int(film['vote_average'])
+            date_str = film['release_date'].split('-')
+            date_int = [int(element) for element in date_str]
+            release_date = date(date_int[0], date_int[1], date_int[2]) + timedelta(weeks=10, days=4)
+            example['Last_day_in_cinema'] = release_date.strftime('%Y-%m-%d')
+            self.collection_of_structures.append(example)
+        return self.collection_of_structures
+
+    def write_collection_to_file(self, path):
+        with open(path, 'w', newline='') as csv_file:
+            writer = csv.writer(csv_file)
+            writer.writerow(['Data'])
+            writer.writerows(data)
+
+
+def change_genre_id(movie):
+    movie['genre_ids'][0] = 22
+    return movie
```

Comment:
move it next to **def replace_genre(self):**
because there it placed out of the context